### PR TITLE
add ccEditionsCount

### PR DESCRIPTION
**Pull request by mattt-w**
editionsCount contains both shared contract editions (v1,v2,v3) and creator contract editions (v4)

Create a new artist field called ccEditionsCount to count v4 editions

Use existing editionsCount field to count v1,v2,v3 editions

### CODE REVIEW

File: src/services/Artist.service.ts [hunk 1]
```diff
@@ -43,9 +44,14 @@ export function loadOrCreateArtist(address: Address): Artist {
     return artist as Artist;
 }
 
-export function addEditionToArtist(artistAddress: Address, editionNumber: string, totalAvailable: BigInt, created: BigInt): Artist {
+export function addEditionToArtist(artistAddress: Address, editionNumber: string, totalAvailable: BigInt, created: BigInt, version: string): Artist {
```
Comment (jamesmorgan) [inline src/services/Artist.service.ts:47]: Is this kodaVersion - if so we have some constants to pass in to the function to make it clearer. `KodaVersions.ts`

File: src/services/Artist.service.ts [hunk 2]
```diff
@@ -43,9 +44,14 @@ export function loadOrCreateArtist(address: Address): Artist {
...
     let artist = loadOrCreateArtist(artistAddress)
-    artist.editionsCount = artist.editionsCount.plus(ONE)
+    
+    if (version === '4') {
+        artist.ccEditionsCount = artist.ccEditionsCount.plus(ONE)
```
Comment (jamesmorgan) [inline src/services/Artist.service.ts:51]: `editionsCount` is used in other services as a data point - need to work out if we still want to split this IMO

One option is we keep `editionsCount` as the total count and then have a separate count for just CC?

**Review by jamesmorgan — CHANGES_REQUESTED** [review]

File: src/services/Artist.service.ts (hunk 1 above)
Comment (mattt-w) [reply src/services/Artist.service.ts]: changed to consts in KodaVersions.ts

File: src/services/Artist.service.ts (hunk 2 above)
Comment (mattt-w) [reply src/services/Artist.service.ts:52]: Agree that splitting `editionsCount` could affect other services.

Have made `editionsCount` the total edition amount and `ccEditionsCount` the total edition amount deployed on creator contracts

Can just subtract `ccEditionsCount from editionsCount` on the front end to get split counts

**Review by jamesmorgan — APPROVED** [review]

**Pull request merged**